Sources/swift-package-editor/SwiftPackageEditorTool.swift

```diff
@@ -0,0 +1,345 @@
+        abstract: "Add a dependency to the current package.")
+
+    @Argument(help: "The URL of a remote package, or the path to a local package")
+    var dependencyURL: String
+
+    @Option(help: "Specifies an exact package version requirement")
+    var exact: Version?
+
+    @Option(help: "Specifies a package revision requirement")
+    var revision: String?
+
+    @Option(help: "Specifies a package branch requirement")
+    var branch: String?
+
+    @Option(help: "Specifies a package version requirement from the specified version up to the next major version")
+    var from: Version?
+
+    @Option(help: "Specifies a package version requirement from the specified version up to the next minor version")
+    var upToNextMinorFrom: Version?
+
+    @Option(help: "Specifies the upper bound of a range-based package version requirement")
+    var to: Version?
+
+    @Option(help: "Specifies the upper bound of a closed range-based package version requirement")
+    var through: Version?
+
+    func run(_ editorTool: EditorTool) throws {
+        var requirements: [PackageDependencyRequirement] = []
+        if let exactVersion = exact {
+            requirements.append(.exact(exactVersion.description))
+        }
+        if let revision = revision {
+            requirements.append(.revision(revision))
+        }
+        if let branch = branch {
+            requirements.append(.branch(branch))
+        }
+        if let version = from {
+            requirements.append(.upToNextMajor(version.description))
+        }
+        if let version = upToNextMinorFrom {
+            requirements.append(.upToNextMinor(version.description))
+        }
+
+        guard requirements.count <= 1 else {
+            editorTool.diagnostics.emit(.error("only one requirement is allowed when specifiying a dependency"))
+            throw ExitCode.failure
+        }
+
+        var requirement = requirements.first
+
+        if case .upToNextMajor(let rangeStart) = requirement {
+            guard to == nil || through == nil else {
+                editorTool.diagnostics.emit(.error("'--to' and '--through' may not be used in the same requirement"))
+                throw ExitCode.failure
+            }
+            if let rangeEnd = to {
+                requirement = .range(rangeStart.description, rangeEnd.description)
+            } else if let closedRangeEnd = through {
+                requirement = .closedRange(rangeStart.description, closedRangeEnd.description)
+            }
+        } else {
+            guard to == nil, through == nil else {
+                editorTool.diagnostics.emit(.error("'--to' and '--through' may only be used with '--from' to specify a range requirement"))
+                throw ExitCode.failure
+            }
+        }
+
+        do {
+            try editorTool.packageEditor.addPackageDependency(url: dependencyURL, requirement: requirement)
+        } catch Diagnostics.fatalError {
```

Comment:
I think we want to get away from Diagnostics.fatalError, its still in place for partial manifest loading but in this case it seems like we can just throw at the call site instead of producing diagnostics then throwing Diagnostics.fatalError